src/compiler/checker.ts

```diff
@@ -17419,9 +17419,12 @@ namespace ts {
                     }
                 }
 
-                checkCollisionWithCapturedThisVariable(node, node.name);
-                checkCollisionWithRequireExportsInGeneratedCode(node, node.name);
-                checkCollisionWithGlobalPromiseInGeneratedCode(node, node.name);
+                if (isIdentifier(node.name)) {

```

Comment:
Was this a bug, or are we just trying to assert here that this is an `Identifier`?